src/time_integration/paired_explicit_runge_kutta/polynomial_optimizer.jl

```diff
@@ -0,0 +1,117 @@
+using LinearAlgebra: eigvals
+using ECOS
+using Convex
+const MOI = Convex.MOI
+
+function filter_eigvals(eig_vals, threshold)
+    filtered_eigvals_counter = 0
```

Comment:
One thing we should also do: 

1. Remove eigenvalues with positive real part (as for these "stability" makes no sense)
2. Remove eigenvalues with negative imaginary part (due to symmetry around the real axis it suffices to consider only the eigenvalues with negative or positive imaginary part